src/SprykerEco/Zed/Unzer/Business/Payment/Saver/UnzerPaymentSaver.php

```diff
@@ -0,0 +1,201 @@
+
+use Exception;
+use Generated\Shared\Transfer\MerchantResponseTransfer;
+use Generated\Shared\Transfer\MerchantTransfer;
+use Generated\Shared\Transfer\MerchantUnzerParticipantTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\SaveOrderTransfer;
+use Generated\Shared\Transfer\UnzerPaymentTransfer;
+use SprykerEco\Shared\Unzer\UnzerConfig as UnzerSharedConfig;
+use SprykerEco\Zed\Unzer\Business\Payment\Mapper\UnzerPaymentMapperInterface;
+use SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface;
+use SprykerEco\Zed\Unzer\Business\Writer\UnzerWriterInterface;
+use SprykerEco\Zed\Unzer\UnzerConfig;
+
+class UnzerPaymentSaver implements UnzerPaymentSaverInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+     */
+    protected $unzerReader;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Writer\UnzerWriterInterface
+     */
+    protected $unzerWriter;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\UnzerConfig
+     */
+    protected $unzerConfig;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Payment\Mapper\UnzerPaymentMapperInterface
+     */
+    protected $unzerPaymentMapper;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     * @param \SprykerEco\Zed\Unzer\Business\Writer\UnzerWriterInterface $unzerWriter
+     * @param \SprykerEco\Zed\Unzer\UnzerConfig $unzerConfig
+     * @param \SprykerEco\Zed\Unzer\Business\Payment\Mapper\UnzerPaymentMapperInterface $unzerPaymentMapper
+     */
+    public function __construct(
+        UnzerReaderInterface $unzerReader,
+        UnzerWriterInterface $unzerWriter,
+        UnzerConfig $unzerConfig,
+        UnzerPaymentMapperInterface $unzerPaymentMapper
+    ) {
+        $this->unzerReader = $unzerReader;
+        $this->unzerWriter = $unzerWriter;
+        $this->unzerConfig = $unzerConfig;
+        $this->unzerPaymentMapper = $unzerPaymentMapper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\SaveOrderTransfer $saveOrderTransfer
+     *
+     * @throws \Exception
+     *
+     * @return void
+     */
+    public function saveOrderPayment(QuoteTransfer $quoteTransfer, SaveOrderTransfer $saveOrderTransfer): void
+    {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== UnzerSharedConfig::PROVIDER_NAME) {
+            return;
+        }
+
+        $paymentUnzerTransfer = $this->unzerReader->getPaymentUnzerByOrderReference($saveOrderTransfer->getOrderReference());
+        if ($paymentUnzerTransfer->getIdPaymentUnzer() !== null) {
+            //@todo refactor
```

Comment:
Imho, it would be great if the @todo is solved or at least it says what exactly should be refactored.